projects/js-toolkit/packages/portal-base/src/build/themeSpritemap.ts

```diff
@@ -0,0 +1,172 @@
+/**
+ * SPDX-FileCopyrightText: © 2021 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+/* eslint-disable @liferay/no-dynamic-require */
+
+import {
+	ClientExtensionConfigJson,
+	FilePath,
+	Project,
+	ThemeSpritemapBuildOptions,
+	format,
+} from '@liferay/js-toolkit-core';
+import fs from 'fs';
+import globby from 'globby';
+import JSZip from 'jszip';
+import path from 'path';
+
+import abort from '../util/abort';
+
+const {error, info, print} = format;
+
+const HEADER_REGEXP = /<!--(.*)-->/s;
+
+const SPRITEMAP_FILE_NAME = 'spritemap.svg';
+
+export default async function customElement(project: Project): Promise<void> {
+	const {enableSVG4Everybody = false, extendClay = false} = project.build
+		.options as ThemeSpritemapBuildOptions;
+
+	fs.mkdirSync(project.build.dir.asNative, {recursive: true});
+
+	print(info` Generating Spritemap:`);
+
+	if (!fs.existsSync(project.srcDir.asNative)) {
+		abort(`Source directory, (${project.srcDir.asNative}), doesn't exist.`);
+	}
+
+	let svgContent =
+		'<?xml version="1.0" encoding="UTF-8"?>' +
+		'<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">' +
+		'<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">';
+
+	if (extendClay) {
+		const clayPath = require.resolve('@clayui/css');
+
+		if (!clayPath) {
+			print(
+				error` @clay/css package not found. Add as a dependency to your client extension via npm.`
+			);
+		}
+
+		const claySpritemapPath = require.resolve(
+			'@clayui/css/lib/images/icons/icons.svg'
```

Comment:
is this going to be there forever or could it change depending on clay's version?

If it can change we may need to move it to the config and provide a default by inspecting the project (or hard coded in first place).

Reply:
I think this will always be there, or at least it hasn't changed in the entire history of clay.

Reply:
:+1: